### ISSUE TITLE
BAHOZ - Attacker can drain all funds in the wallet using a flashloan

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-05-dodo-judging#36]
BAHOZ

high

# Attacker can drain all funds in the wallet using a flashloan

## Summary

Loan `_initiator` is not checked in `executeOperation()` and can be exploited to steal all funds in the contract

## Vulnerability Detail

`executeOperation()` is called by the lending pool and it is gated by `onlyLendingPool` modifier. However, anyone can request a flash loan from lending pool with any contract as receiver and exploit this function to steal all funds in the contract.

An attacker can request a flash loan on victims address and set swap params to make a callback to his own contract and steal all the funds in the victims contract. Since there is no check for `_initiator` in `executeOperation()`, there is no effective access control.

## Impact

Critical Impact, all funds in all contracts can easily be stolen.

## Code Snippet

https://github.com/sherlock-audit/2023-05-dodo/blob/main/dodo-margin-trading-contracts/contracts/marginTrading/MarginTrading.sol#L121-L127

## Tool used

Manual Review

## Recommendation

Check the initiator of the flash loan and revert if the initiator is not authorized.

Duplicate of #34
